### PR TITLE
gh-84131: Remove the deprecated pathlib.Path.link_to method.

### LINKED ISSUE #84131: Add pathlib.Path.hardlink_to()
BPO | [39950](https://bugs.python.org/issue39950)
--- | :---
Nosy | @gpshead, @miss-islington, @tirkarthi, @barneygale, @escape0707
PRs | <li>python/cpython#18909</li><li>python/cpython#26155</li><li>python/cpython#26158</li><li>python/cpython#26178</li>

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = <Date 2021-06-13.15:49:54.023>
created_at = <Date 2020-03-13.00:50:05.346>
labels = ['type-feature', 'library', '3.10']
title = 'Add pathlib.Path.hardlink_to()'
updated_at = <Date 2021-06-13.15:49:54.022>
user = 'https://github.com/barneygale'
```

bugs.python.org fields:
```python
activity = <Date 2021-06-13.15:49:54.022>
actor = 'barneygale'
assignee = 'none'
closed = True
closed_date = <Date 2021-06-13.15:49:54.023>
closer = 'barneygale'
components = ['Library (Lib)']
creation = <Date 2020-03-13.00:50:05.346>
creator = 'barneygale'
dependencies = []
files = []
hgrepos = []
issue_num = 39950
keywords = ['patch']
message_count = 9.0
messages = ['364060', '365275', '385199', '385458', '385471', '385475', '391734', '392779', '393765']
nosy_count = 5.0
nosy_names = ['gregory.p.smith', 'miss-islington', 'xtreak', 'barneygale', 'escape0707']
pr_nums = ['18909', '26155', '26158', '26178']
priority = 'normal'
resolution = 'fixed'
stage = 'resolved'
status = 'closed'
superseder = None
type = 'enhancement'
url = 'https://bugs.python.org/issue39950'
versions = ['Python 3.10']
```

</p></details>

### PR DESCRIPTION
**Pull request by gpshead**

### CODE REVIEW

File: Doc/whatsnew/3.8.rst [hunk 1]
```diff
@@ -1087,6 +1087,9 @@ contain characters unrepresentable at the OS level.
 Added :meth:`pathlib.Path.link_to()` which creates a hard link pointing
 to a path.
 (Contributed by Joannah Nanjekye in :issue:`26978`)
+Note that ``link_to`` was deprecated in 3.10 and removed in 3.12 in
```
Comment (JelleZijlstra) [inline Doc/whatsnew/3.8.rst:1090]: Maybe we should backport this message to the other bugfix branches?

File: Misc/NEWS.d/next/Library/2022-05-08-19-21-14.gh-issue-84131.rG5kI7.rst [hunk 2]
```diff
@@ -0,0 +1,3 @@
+The :class:`pathlib.Path` deprecated method ``link_to`` has been removed.
+Use 3.10's :meth:`pathlib.Path.hardlink_to` method instead as its semantics
+are consistent with that of :meth:`pathlib.Path.symlink_to`.
```
Comment (barneygale) [inline Misc/NEWS.d/next/Library/2022-05-08-19-21-14.gh-issue-84131.rG5kI7.rst]:
```suggestion
Use 3.10's :meth:`~pathlib.Path.hardlink_to` method instead as its semantics
are consistent with that of :meth:`~pathlib.Path.symlink_to`.
```

Adds tilde (`~`) characters to suppress the `pathlib.Path.` prefix in the link text, as we've already introduced the `pathlib.Path` namespace in the first line.

**Review by barneygale — APPROVED** [review]
Possibly tiny nit on the `NEWS` entry, otherwise looks good to me!

File: Doc/whatsnew/3.8.rst (hunk 1 above)
Comment (gpshead) [reply Doc/whatsnew/3.8.rst:1090]: Agreed, I'll do release branch PRs for that after this merges.

**Review by brettcannon — APPROVED** [review]

**Pull request merged**